### DISCUSSION TITLE
Some problems in topi.nn.matmul() code

### DISCUSSION THREAD

**Discussion by YuChenHAHAHA** [mlc-ai/mlc-en#30] [General]
First of all I want to thank you for opening this course.
When I did assignment 1, I found a problem in "tvm.topi.nn.matmul(tensor_a, tensor_b)". 
The script defined the shape of tensor_b is [out_dim, in_dim].
![Screenshot from 2022-07-21 17-04-46](https://user-images.githubusercontent.com/56185866/180240068-c326d175-374b-4818-a74f-04c22e1a233e.png)

However, in the function "matmul()" in "dense.py" in TVM (line 83), the definition of tensor_b's shape is contrary. 
This will result in an error on line 86.
![Screenshot from 2022-07-21 17-00-35](https://user-images.githubusercontent.com/56185866/180240183-b42cf828-ddae-4e68-8512-9137136e66c9.png)

Thank you again for everything you’ve done.

**Comment by jinhongyii** [comment]
There may be something wrong with the document. If you take a look at the python implementation of this function, you will find that `transpose_b = True` corresponds to tensor_b's input shape `[out_dim, in_dim]`, while `transpose_b = False` corresponds to tensor_b's input shape `[in_dim, out_dim]`. So, in your case, you should pass `transpose_b = True` as argument. 

By the way, there's another API tvm.topi.nn.dense which is a wrapper for tvm.topi.nn.matmul. It is equivalent to tvm.topi.nn.matmul with `transpose_a = False` and `transpose_b = True`. So you can also use it here.
